test/kernels/test_fidelity_qkernel.py

```diff
@@ -106,10 +106,26 @@ def test_defaults(self):
 
         self.assertGreaterEqual(score, 0.5)
 
+    def test_max_circuits_per_job(self):
+        """Test max_circuits_per_job parameters."""
+        kernel_all = FidelityQuantumKernel(feature_map=self.feature_map, max_circuits_per_job=None)
+        kernel_matrix_all = kernel_all.evaluate(x_vec=self.sample_train)
+
+        kernel_more = FidelityQuantumKernel(feature_map=self.feature_map, max_circuits_per_job=20)
```

Comment:
Generally, if the unit test is more a set of independent tests, its better to have the test such that if one fails the others still get tested. Here is the assert kernel_more fails it will stop there. In this case if you use with subtest, you can see examples here https://github.com/qiskit-community/qiskit-machine-learning/blob/42da2e6c1abd3a92494ddd81159112c53cde0c67/test/circuit/library/test_qnn_circuit.py#L72, this would be better

e.g

```
with self.subTest("20 circuits per job"):
    kernel_more = FidelityQuantumKernel(feature_map=self.feature_map, max_circuits_per_job=20)
    kernel_matrix_more = kernel_more.evaluate(x_vec=self.sample_train)
   np.testing.assert_equal(kernel_matrix_all, kernel_matrix_more)  
     
with self.subTest("1 circuit per job)
  ...
```
And if the test logic and the assert are kept together then I think its clearer/cleaner.

In other places we use ddt and its decorators to parameterize a unit test which could be done here so a list of values for different sizes could be passed in (ddt decorator will end up making individual tests) but here maybe the reference kernel build each time would be less efficient and while one could find ways around that for this case I think subTest wins out.

Reply:
I thought about efficiency, however, `self.subTest` idea sounds better. Thanks, I will make the changes.

Reply:
I have made the changes, please let me know if it requires any further changes. 

https://github.com/OkuyanBoga/hc-qiskit-machine-learning/blob/c330f793293cf1c7f353c39369dc7534e0a07eed/test/kernels/test_fidelity_qkernel.py#L111-L122